dags/oss_know/libs/util/base.py

```diff
@@ -52,3 +59,105 @@ def get_redis_client(redis_client_info):
+    if emaildomain in COMPANY_COUNTRY.keys():
+        return COMPANY_COUNTRY[emaildomain]
+    else:
+        return None
+
+def infer_company_from_emaildomain(email):
+    """
+        :param  company: the company given by github
+        :return country_name  : the english name of a country
+        """
+    emaildomain = str(re.findall(r"@(.+?)\.", email))
+    if emaildomain in COMPANY_COUNTRY.keys():
+        return emaildomain
+    else:
+        return None
+
+
+def infer_country_from_location(githubLocation):
+    """
+        :param  githubLocation: the location given by github
+        :return country_name  : the english name of a country
+        """
+    from airflow.models import Variable
+    api_token = Variable.get(LOCATIONGEO_TOKEN, deserialize_json=True)
+    geolocator = GoogleV3(api_key=api_token)
+    geo_res = geolocator.geocode(githubLocation, language='en')
+    if geo_res:
+        return geo_res.address.split(',')[-1].strip()
+    return None
+
+
+def infer_all_info_from_location(githubLocation):
+    """
+           :param  githubLocation: the location given by github
+           :return country_name  : the english name of a country
+           """
+    from airflow.models import Variable
+    api_token = Variable.get(LOCATIONGEO_TOKEN, deserialize_json=True)
+    geolocator = GoogleV3(api_key=api_token)
+    geo_res = geolocator.geocode(githubLocation, language='en')
+    if geo_res:
+        return geo_res.address
+    return None
+
+
+
+def infer_country_from_company(company):
+    """
+        :param  company: the company given by github
+        :return country_name  : the english name of a country
+        """
+    company = company.replace("@", " ").lower().strip()
+    company_country = CIMultiDict(COMPANY_COUNTRY)
+    if company in company_country.keys():
+        return company_country[company]
+    return None
+
+
+def infer_country_insert_into_profile(latest_github_profile):
+    try:
+        inferiors = [
+            ("country_inferred_from_email_cctld", "email", infer_country_from_emailcctld),
+            ("country_inferred_from_email_domain_company", "email", infer_country_from_emaildomain),
+            ("country_inferred_from_location", "location", infer_country_from_location),
+            ("country_inferred_from_company", "company", infer_country_from_company),
+            ("company_inferred_from_email_domain_company", "email", infer_company_from_emaildomain),
+            ("inferred_from_location", "location", infer_all_info_from_location)
+        ]
+
+        for tup in inferiors:
+            key, original_key, infer = tup
+            original_property = latest_github_profile[original_key]
+            latest_github_profile[key] = infer(original_property) if original_property else None
+
+    except (urllib3.exceptions.MaxRetryError, requests.exceptions.ProxyError) as error:
+        logger.error(f"error occurs when inferring country, {error}")
```

Comment:
Why not just loop the inferiors? Repeating similarities manually is always bad practice in software engineering.